src/proxy_app/health_checker.py

```
@@ -102,6 +102,26 @@ async def check_single_provider(
+
+                is_healthy = status.get("status") == "healthy"
+                latency = status.get("avg_latency_ms")
+                get_telemetry_manager().update_provider_health(
+                    provider=provider_name,
+                    model=status.get("model_used"),
+                    is_healthy=is_healthy,
+                    failure_rate=0.0 if is_healthy else 1.0,
+                )
+                if latency is not None:
+                    _ = latency  # available for future latency-aware scoring
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)